### PR TITLE
Added support for Visual Studio code.

### PR DESCRIPTION
**Pull request by Mailaender**
This is the counterpart to https://github.com/OpenRA/OpenRA/pull/18280.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            "linux": {
+                "type": "mono"
+            },
+            "osx": {
+                "type": "mono"
+            },
+            "request": "launch",
+            "program": "${workspaceRoot}/engine/OpenRA.Game.exe",
+            "cwd": "${workspaceRoot}",
+            "args": ["Engine.LaunchPath=${workspaceRoot}", "Engine.ModSearchPaths=${workspaceRoot}/mods", "Game.Mod=ra2", "Debug.DisplayDeveloperSettings=true"]
```
Comment (abcdefg30) [inline .vscode/launch.json:16]: Shouldn't the `ModSearchPaths` include just `./mods` here as well? (See https://github.com/OpenRA/ra2/blob/master/launch-game.sh#L21)

**Review by abcdefg30 — COMMENTED** [review]
`launch.json` and `tasks.json` should be converted to use tabs instead of spaces to be consistent with the other files. Lgtm otherwise.

File: .vscode/launch.json (hunk 1 above)
Comment (Mailaender) [reply .vscode/launch.json:16]: `${workspaceRoot}/mods` probably translates to just `./mods` already.

**Comment by Mailaender** [comment]
Nope, these are (Visual Studio Code) `.json` files. Just like `.travis.yml` it should be indented in upstream way.

**Comment by abcdefg30** [comment]
Half of the files here are already indented using tabs, and OpenRA/OpenRA#18280 has all files indented with tabs as well.

File: .vscode/launch.json (hunk 1 above)
Comment (abcdefg30) [reply .vscode/launch.json:16]: `./mods` will be in the engine directory, won't it?

**Comment by Mailaender** [comment]
Okay, if tabs don't count as syntax errors then here you go.

File: .vscode/launch.json (hunk 1 above)
Comment (Mailaender) [reply .vscode/launch.json]: I think that is relative to the repository root so it will be the `mods` folder where `ra2` is in.

Comment (abcdefg30) [reply .vscode/launch.json]: Then we'd be missing the mods folder of the engine directory or is that included in some other way?

Comment (Mailaender) [reply .vscode/launch.json]: I can only say that is the exact same setup I use for @OpenHV and I never encountered problems during debugging.

Comment (abcdefg30) [reply .vscode/launch.json]: I'm just wondering since we include that in the start scripts (see link in the first comment). Perhaps the `modcontent` mod?

Comment (Mailaender) [reply .vscode/launch.json]: Maybe. I never debugged during content installation.

Comment (abcdefg30) [reply .vscode/launch.json]: What is the status here. Does this work with the content mod or not?

Comment (Mailaender) [reply .vscode/launch.json]: Amended it so it does work now.

**Review by abcdefg30 — APPROVED** [review]
Thanks!

**Pull request merged**